ground/src/main/java/com/google/android/ground/ui/common/EphemeralPopups.kt

```diff
@@ -16,25 +16,75 @@
 package com.google.android.ground.ui.common
 
 import android.app.Application
+import android.view.View
 import android.widget.Toast
 import androidx.annotation.StringRes
 import com.google.android.ground.R
+import com.google.android.material.snackbar.Snackbar
 import javax.inject.Inject
 import javax.inject.Singleton
 
 /** Displays short-lived messages such as toasts that are shown over other UI elements. */
+@Suppress("UseDataClass")
 @Singleton
 class EphemeralPopups @Inject constructor(private val context: Application) {
 
-  fun showError(@StringRes messageId: Int) = showLong(messageId)
+  enum class PopupDuration {
+    SHORT,
+    LONG,
+    INDEFINITE,
+  }
 
-  fun showError(message: String) = showLong(message)
+  /** Defines functions to render a popup that displays an error message to the user. */
+  inner class ErrorPopup {
+    fun show(@StringRes messageId: Int, duration: PopupDuration = PopupDuration.LONG) =
+      showToast(messageId, duration)
 
-  // TODO: Rename to unknownError?
-  fun showError() = showLong(R.string.unexpected_error)
+    fun show(message: String, duration: PopupDuration = PopupDuration.LONG) =
+      showToast(message, duration)
 
-  private fun showLong(@StringRes messageId: Int) =
-    Toast.makeText(context, messageId, Toast.LENGTH_LONG).show()
+    fun unknownError() = showToast(R.string.unexpected_error, duration = PopupDuration.LONG)
 
-  private fun showLong(message: String) = Toast.makeText(context, message, Toast.LENGTH_LONG).show()
+    private fun showToast(@StringRes messageId: Int, duration: PopupDuration) {
+      val dur =
+        if (duration == PopupDuration.SHORT) {
+          Toast.LENGTH_SHORT
+        } else {
+          Toast.LENGTH_LONG
+        }
+      Toast.makeText(context, messageId, dur).show()
+    }
```

Comment:
We can get rid of this helper method by reusing the other one. Something like:

`showToast(context.getString(messageId), duration)`